File: client-ios/Sources/Core/Utils/RoomStatuses.swift
import Foundation
import SerenadaCore

enum RoomStatusIndicatorState: Equatable {
    case hidden
    case waiting
    case full
}

enum RoomStatuses {
    static func indicatorState(for occupancy: RoomOccupancy?) -> RoomStatusIndicatorState {
        let count = occupancy?.count ?? 0
        if count <= 0 {
            return .hidden
        }

        let maxParticipants: Int
        if let max = occupancy?.maxParticipants, max >= 2 {
            maxParticipants = max
        } else {
            maxParticipants = 2
        }

        return count >= maxParticipants ? .full : .waiting
    }
}

File: client-android/app/src/main/java/app/serenada/android/call/RoomStatuses.kt
package app.serenada.android.call

import app.serenada.core.RoomOccupancy

enum class RoomStatusIndicatorState {
    Hidden,
    Waiting,
    Full,
}

object RoomStatuses {
    fun indicatorState(status: RoomOccupancy?): RoomStatusIndicatorState {
        val count = status?.count ?: 0
        if (count <= 0) {
            return RoomStatusIndicatorState.Hidden
        }

        val maxParticipants = status?.maxParticipants
        val normalizedMaxParticipants =
            if (maxParticipants != null && maxParticipants >= 2) {
                maxParticipants
            } else {
                2
            }

        return if (count >= normalizedMaxParticipants) {
            RoomStatusIndicatorState.Full
        } else {
            RoomStatusIndicatorState.Waiting
        }
    }
}
